supabase/functions/process-transcripts/index.ts

```diff
@@ -0,0 +1,160 @@
+            *** Topics must be a list of topics that were discussed in the transcript, include topics not mentioned but that relate to the topics discussed.\n\n
+             `,
+                },
+            ],
+            response_format: { type: 'json_object' },
+        });
+
+        const responseData = JSON.parse(
+            response.data.choices[0].message.content
+        );
+        console.log('Response Data:', responseData);
+
+        const { summary, topics } = responseData;
+
+        // Create meta data for each indivual transcript
+        let transcript_and_metadata: Object = {};
+        for (const record of data) {
+            const response = await openaiClient.chat.completions.create({
+                model: 'gpt-4-0125-preview',
+                messages: [
+                    {
+                        role: 'system',
+                        content: `
+                        This is a transcript that contains information about your user.\n 
+                        Your task is to organize the information in a way that makes sense to you.\n
+                        Your response must be in json format with the three following keys: "verbose_version", "topics".
+                        `,
+                    },
+                    {
+                        role: 'user',
+                        content: `${record.raw_text}\n\nGiven the information about the user,
+                        provide a verbose version of the transcript, and the topics discussed\n.
+                        *** Verbose version must include all the information in the original transcript, but in a more verbose manner.\n
+                        *** Topics must be a list of topics that were discussed in the transcript, include topics not mentioned but that relate to the topics discussed.\n\n
+                        `,
+                    },
+                ],
+                response_format: { type: 'json_object' },
+            });
+
+            const responseData = JSON.parse(
+                response.data.choices[0].message.content
+            );
+            console.log('Response Data:', responseData);
+
+            transcript_and_metadata[record.id] = {
+                raw_text: record.raw_text,
+                verbose_version: responseData.verbose_version,
+                topics: responseData.topics,
+                summary: `This is an summary of the broader conversation so you have more context ${summary}`,
+            };
+        }
+
+        // Embed the individual transcript with their metadata
+        let flattenedData = '';
+        for (const [id, metadata] of Object.entries(transcript_and_metadata)) {
+            flattenedData += `Raw Text: ${
+                metadata.raw_text
+            }, Verbose Version: ${
+                metadata.verbose_version
+            }, Topics: ${metadata.topics.join(', ')} ${topics.join(
+                ', '
+            )}, Summary: ${metadata.summary}.`;
+
```

Comment:
I'd probably store some of the metadata separately from the raw text, since if you concatenate everything it kind of loses the purpose, since you lose ability to search explicitly by topics/summary.

Reply:
We could store the topics as a field in the database. I need to look into pgvector to see if you can attach metadata to the embedding.

Reply:
For time when we create a record in the database we have a timestamp. These embeddings would update that record and during retrieval we can use that timestamp

Reply:
Would the timestamp correspond to when the embeddings were created (i.e. when this job runs) or when the original text comes through? It would be useful to maintain that original timestamp for the reasons @Krupskis indicated in their other comment.